app/controllers/spree/amazon_controller.rb

```diff
@@ -81,16 +82,23 @@ def complete
     @order = current_order
     authorize!(:edit, @order, cookies.signed[:guest_token])
     complete_amazon_order!
-
+    
     if @order.confirm? && @order.next
       @current_order = nil
       flash.notice = Spree.t(:order_processed_successfully)
       flash[:order_completed] = true
       redirect_to spree.order_path(@order)
     else
+      amazon_transaction = @order.amazon_transaction
       @order.state = 'cart'
-      @order.amazon_transactions.destroy_all
-      redirect_to cart_path, notice: "Unable to process order"
+      if amazon_transaction.soft_decline
+        @order.save!
+        redirect_to address_amazon_order_path, notice: amazon_transaction.message
+      else
+        @order.amazon_transactions.destroy_all
+        @order.save!
+        redirect_to cart_path, notice: "Your payment could not be processed. Please try to place the order again using another payment method."
+      end
```

Comment:
As this is a gem, and not a customer only project, it's better to use i18l translations. It'll shorten the width here as well.